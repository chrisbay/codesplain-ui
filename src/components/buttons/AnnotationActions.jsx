import React, { PropTypes } from 'react';
import { IconButton } from 'material-ui';
import Previous from 'material-ui/svg-icons/navigation/arrow-back';
import Next from 'material-ui/svg-icons/navigation/arrow-forward';
import CloseIcon from 'material-ui/svg-icons/navigation/close';
import EditIcon from 'material-ui/svg-icons/editor/mode-edit';

const styles = {
  container: {
    display: 'inline-flex',
    justifyContent: 'space-between',
    flexGrow: '1',
  },
};

const AnnotationActions = ({
  hasNextAnnotation,
  getNextAnnotation,
  hasPrevAnnotation,
  getPreviousAnnotation,
  onClose,
  onEdit,
  disableEdit,
}) => (
  <div style={styles.container}>
    <IconButton
      tooltip="Edit"
      tooltipPosition="top-right"
      onTouchTap={onEdit}
      disabled={disableEdit}
    >
      <EditIcon />
    </IconButton>
    <div>
      <IconButton
        disabled={!hasPrevAnnotation}
        id="previous-annotation"
        onTouchTap={getPreviousAnnotation}
        tooltip="Previous Annotation"
        tooltipPosition="top-right"
        touch
      >
        <Previous />
      </IconButton>
      <IconButton
        disabled={!hasNextAnnotation}
        id="next-annotation"
        onTouchTap={getNextAnnotation}
        tooltip="Next Annotation"
        tooltipPosition="top-right"
        touch
      >
        <Next />
      </IconButton>
      <IconButton
        tooltip="Close"
        tooltipPosition="top-right"
        onTouchTap={onClose}
      >
        <CloseIcon />
      </IconButton>
    </div>
  </div>
  );

AnnotationActions.propTypes = {
  hasNextAnnotation: PropTypes.bool.isRequired,
  hasPrevAnnotation: PropTypes.bool.isRequired,
  getNextAnnotation: PropTypes.func.isRequired,
  getPreviousAnnotation: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  disableEdit: PropTypes.bool,
};

AnnotationActions.defaultProps = {
  disableEdit: false,
};

export default AnnotationActions;
