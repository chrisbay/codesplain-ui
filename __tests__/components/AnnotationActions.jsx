import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import AnnotationActions from '../../src/components/buttons/AnnotationActions';

const defaultProps = {
  getNextAnnotation: jest.fn(),
  getPreviousAnnotation: jest.fn(),
  hasNextAnnotation: false,
  hasPrevAnnotation: false,
  onClose: jest.fn(),
  onEdit: jest.fn(),
};

describe('<AnnotationPanel />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshots', () => {
    it('matches snapshot', () => {
      const wrapper = shallowWithContext(<AnnotationActions {...defaultProps} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
