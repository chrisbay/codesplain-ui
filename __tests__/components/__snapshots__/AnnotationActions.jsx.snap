exports[`<AnnotationPanel /> snapshots matches snapshot 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
      "flexGrow": "1",
      "justifyContent": "space-between",
    }
  }>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onTouchTap={[Function]}
    tooltip="Edit"
    tooltipPosition="top-right"
    touch={false}>
    <EditorModeEdit />
  </IconButton>
  <div>
    <IconButton
      disableTouchRipple={false}
      disabled={true}
      iconStyle={Object {}}
      id="previous-annotation"
      onTouchTap={[Function]}
      tooltip="Previous Annotation"
      tooltipPosition="top-right"
      touch={true}>
      <NavigationArrowBack />
    </IconButton>
    <IconButton
      disableTouchRipple={false}
      disabled={true}
      iconStyle={Object {}}
      id="next-annotation"
      onTouchTap={[Function]}
      tooltip="Next Annotation"
      tooltipPosition="top-right"
      touch={true}>
      <NavigationArrowForward />
    </IconButton>
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      onTouchTap={[Function]}
      tooltip="Close"
      tooltipPosition="top-right"
      touch={false}>
      <NavigationClose />
    </IconButton>
  </div>
</div>
`;
